src/nemo-dbus/connection.cpp

```diff
@@ -45,11 +45,13 @@ ConnectionData::ConnectionData(const QDBusConnection &connection, const QLogging
     , m_logs(logs)
 {
     if (connection.isConnected()) {
-        qCDebug(logs, "Connected to %s", qPrintable(connection.name()));
+        QLoggingCategory logCat(logs().categoryName());
+        qCDebug(logCat, "Connected to %s", qPrintable(connection.name()));
```

Comment:
The repeated recreation of the logging category looks suspicious and I'm not either following what's the deprecated api to avoid. The qCDebug etc are the newer logging api and what I can tell it looks the same on Qt6 too.

Reply:
> The repeated recreation of the logging category looks suspicious and I'm not either following what's the deprecated api to avoid. The qCDebug etc are the newer logging api and what I can tell it looks the same on Qt6 too.

@Tofee any thoughts here?

Reply:
> The repeated recreation of the logging category looks suspicious and I'm not either following what's the deprecated api to avoid. The qCDebug etc are the newer logging api and what I can tell it looks the same on Qt6 too.

I can clean the recreation up a bit. The error we're seeing with Qt6 is the following
```
| ../../../git/src/nemo-dbus/connection.cpp: In member function 'QDBusMessage NemoDBus::ConnectionData::blockingCallMethod(const QString&, const QString&, const QString&, const QString&, const QVariantList&)':
| ../../../git/src/nemo-dbus/connection.cpp:158:13: error: invalid use of non-static member function 'const QLoggingCategory& NemoDBus::ConnectionData::logs()'
|   158 |     qCDebug(logs, "DBus invocation (%s %s %s.%s)",
|       |             ^~~~
| ../../../recipe-sysroot/usr/include/QtCore/qloggingcategory.h:168:57: note: in definition of macro 'QT_MESSAGE_LOGGER_COMMON'
|   168 |     for (QLoggingCategoryMacroHolder<level> qt_category(category); qt_category; qt_category.control = false) \
|       |                                                         ^~~~~~~~
| ../../../git/src/nemo-dbus/connection.cpp:158:5: note: in expansion of macro 'qCDebug'
|   158 |     qCDebug(logs, "DBus invocation (%s %s %s.%s)",
|       |     ^~~~~~~
| In file included from ../../../git/src/nemo-dbus/connection.h:38,
|                  from ../../../git/src/nemo-dbus/connection.cpp:33:
| ../../../git/src/nemo-dbus/private/connectiondata.h:115:29: note: declared here
|   115 |     const QLoggingCategory &logs()
|       |                             ^~~~
| In file included from ../../../recipe-sysroot/usr/include/QtCore/QLoggingCategory:1,
|                  from ../../../git/src/nemo-dbus/logging.h:36,
|                  from ../../../git/src/nemo-dbus/connection.cpp:36:

```

Reply:
@pvuorela I cleaned it up a bit now, so there's a few less QLoggingCategory. Happy to squash the 2 commits together.

Reply:
What I would expect is that there wouldn't be a single extra logging category since there's already an instance which was used earlier. Not sure what has changed on the macro declaration to break it.

Wonder if switching from qCDebug(category, const char*, ...) to qCDebug(category) << QString("..").arg() type of usage would avoid this problem. Or then there's also QLoggingCategory::operator() which talks about usage with qCDebug().